workflows/index-generation/ncbi-compress/src/ncbi_compress.rs

```diff
@@ -346,45 +345,87 @@ pub mod ncbi_compress {
+            let mut unique_in_chunk: Vec<(KmerMinHash, &fasta::Record)> = Vec::with_capacity(chunk_size);
+            let mut unique_in_tree_and_chunk: Vec<KmerMinHash> = Vec::with_capacity(chunk_size);
+
+            let chunk = records_iter
+                .borrow_mut()
+                .take(chunk_size)
+                .collect::<Vec<_>>();
+
+            accession_count.add_assign(chunk.len() as u64);
+
+            if chunk.len() == 0 {
+                break;
+            }
+
+            // create signatures for each record in the chunk
             let chunk_signatures = chunk
-                .par_iter()
-                .filter_map(|r| {
-                    let record = r.as_ref().unwrap();
-                    let mut hash;
-                    if is_protein_fasta {
-                        hash = KmerMinHash::new(
-                            scaled,
-                            k,
-                            HashFunctions::murmur64_protein,
-                            seed,
-                            false,
-                            0,
-                        );
-                        hash.add_protein(record.seq()).unwrap();
-                    } else {
-                        hash = KmerMinHash::new(
-                            scaled,
-                            k,
-                            HashFunctions::murmur64_DNA,
-                            seed,
-                            false,
-                            0,
+            .par_iter()
+            .filter_map(|r| {
+                let record = r.as_ref().unwrap();
+                let mut hash;
+                if is_protein_fasta {
+                    hash = KmerMinHash::new(
+                        scaled,
+                        k,
+                        HashFunctions::murmur64_protein,
+                        seed,
+                        false,
+                        0,
+                    );
+                    hash.add_protein(record.seq()).unwrap();
+                    Some((hash, record))
+                } else {
+                    hash = KmerMinHash::new(
+                        scaled,
+                        k,
+                        HashFunctions::murmur64_DNA,
+                        seed,
+                        false,
+                        0,
+                    );
+                    hash.add_sequence(record.seq(), true).unwrap();
+                    Some((hash, record))
+                }
+            }).collect::<Vec<_>>();
+
+            // we need to make sure records within the chunk arn't similar to each other before
+            // we check them against the larger tree
+            for (hash, record) in chunk_signatures {
+                let similar = unique_in_chunk
+                    .par_iter()
+                    .any(|(other, _record)| containment(&hash, &other).unwrap() >= similarity_threshold);
+
+                if !similar {
+                    unique_accession_count.add_assign(1);
```

Comment:
This may still end up being filtered out later we shouldn't increment. We should move this down to the write.

Reply:
fixed